backend-app/controllers/auth_controller.js

```diff
@@ -142,38 +156,35 @@ exports.signup = async (req, res, next) => {
 
 exports.tokenRefresh = async (req, res, next) => {
     try {
-        const { accessToken, refreshToken } = req.body;
+        const { refreshToken } = req.signedCookies;
         if (!refreshToken)
             throw new AppError(400, 'fail', 'Please provide refresh token');
+        const refreshTokenPayload = await AuthUtils.verifyRefreshToken(
+            refreshToken
+        );
+        if (!refreshTokenPayload || !refreshTokenPayload.id)
+            throw new AppError(400, 'fail', 'Invalid refresh token');
+        const user = await User.findById(refreshTokenPayload.id);
+        if (!user) throw new AppError(400, 'fail', 'Invalid refresh token');
+        const accessToken = AuthUtils.generateAccessToken(user._id);
+        //set or override accessToken cookie.
+        AuthUtils.setAccessTokenCookie(res, accessToken);
+        res.sendStatus(204);
+    } catch (err) {
+        next(err);
+    }
+};
+exports.logout = async (req, res, next) => {
+    try {
+        const { accessToken } = req.signCookies;
         if (!accessToken)
             throw new AppError(400, 'fail', 'Please provide access token');
-        const accessTokenPayload = await promisify(jwt.verify)(
-            accessToken,
-            ACCESS_TOKEN_SECRET
-        );
-        const refreshTokenPayload = await promisify(jwt.verify)(
-            refreshToken,
-            REFRESH_TOKEN_SECRET
+        const accessTokenPayload = await AuthUtils.verifyAccessToken(
+            accessToken
         );
-        if (
-            accessTokenPayload?.id?.toString() !==
-            refreshTokenPayload?.id?.toString()
-        )
-            throw new AppError(400, 'fail', 'Invalid access or refresh token');
-        const tokenRecord = await TokenModel.findOne({
-            userId: accessTokenPayload.id,
-            accessToken: accessToken,
-            refreshToken: refreshToken,
-        });
-        //the issued tokens should exist in the database
-        if (!tokenRecord)
-            throw new AppError(400, 'fail', 'Invalid access or refresh token');
-
-        const tokens = await generateTokens(accessTokenPayload.id);
-        res.status(200).json({
-            message: 'Token issued.',
-            tokens,
-        });
+        if (!accessTokenPayload || !accessTokenPayload.id)
```

Comment:
I think you have unfinished business in here, @muttaqin1

Reply:
Okay, i will fix everything tomorrow.

Reply:
sure take your time